x/curating/spec/README.md

```diff
@@ -0,0 +1,224 @@
+
+Only data for posts currently being curated are stored on-chain. Everything else is deleted after the curation window completes and rewards are paid out.
+
+## State
+
+### Parameters
+
+Each of these parameters can be voted on by governance, but will be assigned sane defaults for genesis.
+
+```go
+type Params struct {
+	CurationWindow       	 time.Duration
+	DefaultPostDeposit       sdk.Coin
+	DefaultUpvoteDepost  	 sdk.Coin
+	DefaultModerationDeposit sdk.Coin
+	DefaultVoteAmount        sdk.Coin // amount for 1 vote
+	MaxNumVotes              uint16 // upper-limit on QV voting
+	MaxVendors               uint16 // upper-limit on vendor_ids
+	RewardPoolAllocation 	 sdk.Dec // from inflation
+}
+```
+
+### Base Types
+
+```go
+type Post struct {
+	VendorID        uint64
+	PostID          uint64
+	Creator         sdk.AccAddress
+	RewardAccount	sdk.AccAddress
+	BodyHash        string
+	Deposit         sdk.Coin
+	CurationEndTime time.Time
+}
+```
+
+```go
+type Upvote struct {
+	VendorID        uint64
+	PostID          uint64
+	Curator         sdk.AccAddress
+	VoteAmount      sdk.Coin
+	Deposit         sdk.Coin
+	CurationEndTime time.Time
+}
+```
+
+### Stores
+
+_Stores are KVStores in the multi-store. The key to find the store is the first
+parameter in the list_.
+
+We will use one KVStore `curating` to store two mappings:
+
+- A mapping from `0x01|vendorID|postID` to `Post`.
+- A mapping from `0x02|vendorID|postID|curator` to `Upvote`.
+
+### Post Curation Queue
+
+All queues objects are sorted by timestamp. The time used within any queue is
+first rounded to the nearest nanosecond then sorted. The sortable time format
+used is a slight modification of the RFC3339Nano and uses the the format string
+`"2006-01-02T15:04:05.000000000"`. Notably this format:
+
+- right pads all zeros
+- drops the time zone info (uses UTC)
+
+In all cases, the stored timestamp represents the maturation time of the queue
+element.
+
+For the purpose of tracking the end of post curation windows the post curation queue is kept.
+
+- PostCurationQueue: `0x41 | format(CurationEndTime) -> []VPPair`
+
+```go
+type VPPair struct {
+    VendorID uint64
+    PostID   uint64
+}
+```
+
+During each `EndBlock`, all the posts that have reached the end of their curation window are processed. The upvotes are tallied, and rewards are paid out to via a mix of quadratic voting and quadratic finance matching.
+
+## Messages
+
+### MsgPost
+
+A social media post to be curated is registered on-chain with `MsgPost`. Note that the actual data of the post is not stored, just a hash of the data, and a vendor and post identifier.
+
+Post curation begins as soon as the post is registered on-chain.
+
+```go
+type MsgPost struct {
+	VendorID      uint64 // 1 = twitter, 2 = reddit, etc.
+	PostID        uint64
+	Creator       sdk.AccAddress
+	RewardAccount sdk.AccAddress // optional
+	BodyHash      string // hex string, optional
```

Comment:
Not sure if we should accept the body and the chain produce the hash of this content , reason: different clients may choose to send different hashes . Can't think a way we can make it standard for everyone so its easier to verify .

Reply:
Best to hash on-chain. Mitigates against hash collision attacks as well. Reason I didn't do it initially was because it could increase gas costs if the body was large. But let's go with safety over gas costs for now.